src/servers/graphql-server.ts

```diff
@@ -92,31 +92,42 @@ export const startApolloServer = async ({
         ip = ips
       }
 
-      let wallet, user
+      let wallet
 
       // TODO move from id: uuidv4() to a Jaeger standard
       const logger = graphqlLogger.child({ token, id: uuidv4(), body: context.req?.body })
 
       let domainUser: User | null = null
       return addAttributesToCurrentSpanAndPropagate(
-        { [SemanticAttributes.ENDUSER_ID]: uid, [SemanticAttributes.HTTP_CLIENT_IP]: ip },
+        {
+          [SemanticAttributes.ENDUSER_ID]: userId,
+          [SemanticAttributes.HTTP_CLIENT_IP]: ip,
+        },
         async () => {
-          if (uid) {
-            const loggedInUser = await Users.getUserForLogin({ userId: uid, ip })
-            if (loggedInUser instanceof Error)
+          if (userId) {
+            const lastConnection = new Date()
+
+            const user = await UsersRepository().findByIdAndUpdateLastConnectionDate(
```

Comment:
This function is not appropriate for a `Repository`. Repositories are for retrieving data (`findXxx`) and storing data (aka `persist`) NOT for updating data.
This is (among other things) to prevent coupling of changes of state with the database implementation.
Business logic for around updating state should be implemented on the entities in the domain layer.

Reply:
the previous design requires 2 iteration to mongodb, which is adding latency. 

we don't want to load the state just to update `lastConnection`. `lastConnection` in the context of grqphql is a write only operation, and this can be done in single pass to mongodb.

How would you do that to not incur the additional network latency?

For payment applications, latency is a key element, and requiring additional back and forth to a database to enforce some pattern might not fit the requirement of having fast query.